mautic-inc/plugin-integrations
# MAUT-2700 Fixes issues with filtering/saving sync field mapping

This fixes two issues:
1. Symfony will throw a `The value of type "array" cannot be converted to a valid array key` after updating a field's mapping then trying to filter or click to another page
2. Fields that only support one way would save with a sync direction of `bidirectional`. 

### Issue 1
To reproduce the first issue:
1. Open the SF config page and go to the Lead Field Mapping page
2. Search for "salutation" and map it to a Mautic field
3. Clear the filter
4. Search for "salutation" again and map it to a different Mautic field
5. Click Apply
4. Search for "salutation" again and notice the message/exception `The value of type "array" cannot be converted to a valid array key`

To test the first issue:
Repeat and this time you should not get the message but the filtered results

### Issue 2
To reproduce the second issue:
1. On the Lead Field Mapping, search for the `Converted Contact ID` field and map it to a Mautic field (don't touch the direction drop down). 
2. Click apply 
3. Go the database's plugin_integration_settings table  where name = Salesforce2. Open the field_settings value and look for `ConvertedContactId` in the serialized array. Notice that `syncDirection` is set to `bidirectional`. 

To test the second issue:
1. Repeat but select a different Mautic field and this time the `syncDirection` should be `mautic`.
2. Search for the `Converted Account ID`  and map it to a custom field (again don't touch direction). 
3. Save and look at the database, the sync direction should be `mautic`
